provisioner/ansible/provisioner.go

```diff
@@ -648,15 +648,20 @@ func (p *Provisioner) executeGalaxy(ui packer.Ui, comm packer.Communicator) erro
 		collectionArgs = append(collectionArgs, "-p", filepath.ToSlash(p.config.CollectionsPath))
 	}
 
-	roleInstallError := p.invokeGalaxyCommand(roleArgs, ui, comm)
-	// Return the error if the role installation failed before attempting the collection install
-	if roleInstallError != nil {
+	// Run normal ansible-galaxy install for roles
+	if roleInstallError := p.invokeGalaxyCommand(roleArgs, ui, comm); roleInstallError != nil {
```

Comment:
Are roles always installed or one can fall to the same problem as you had with collections and have only collections set and this fail? 
If one can install only collections I think the same logic should be written for `roles` too, but this one you will have to check for both V1/V2 files type (`src/roles`)

Reply:
I've made some updates to check the `galaxy_file` to make it more clear if only roles, only collections, or both should be installed. The v1 format requirements file will only trigger the roles to install. The v2 format can trigger either or both, based on the keywords included.

I'll hold off on any config changes for now, as I'm hoping they won't be needed.